### ISSUE TITLE
Delete current wallpaper?

### ISSUE THREAD

**Issue by iamgitcat** [Hamza417/Peristyle#103]
Is it possible to make a notification shade button to delete current wallpaper?

Or maybe there is a way to set a wallpaper with intent (I see there are many intents in Peristyle)? I will manage everything else if I could knew which intent and arguments.

Why? I have a script, that periodically runs in Termux and sets random wallpapers with termux-wallpaper from different directories, also writes a path to txt, so other script can delete current wallpaper (from that txt). That makes it easy to sort good and bad wallpapers, I can run scripts from a widget or power menu. The problem is termux-api do not works with my other old device and I search some other way without termux-api. I tried changing wallpaper with `am start -a android.intent.action.ATTACH_DATA \
   -d "file:///sdcard/Download/shontz_valley.jpg" \
   -t "image/*" --grant-read-uri-permission
` and it brings a standard ui menu to manually select home/lock.. That is not cool way, it should just change homescreen wallpaper without any confirmations.

Thanks!

**Comment by Hamza417** [comment]
> Is it possible to make a notification shade button to delete current wallpaper?

Yeah, but you'd need two buttons to do that, one for home screen, one for lock, It can be added in the app to make is easily understandable.

> Or maybe there is a way to set a wallpaper with intent (I see there are many intents in Peristyle)? I will manage everything else if I could knew which intent and arguments.
> 
> Why? I have a script, that periodically runs in Termux and sets random wallpapers with termux-wallpaper from different directories, also writes a path to txt, so other script can delete current wallpaper (from that txt). That makes it easy to sort good and bad wallpapers, I can run scripts from a widget or power menu. The problem is termux-api do not works with my other old device and I search some other way without termux-api. I tried changing wallpaper with am start -a android.intent.action.ATTACH_DATA \ -d "file:///sdcard/Download/shontz_valley.jpg" \ -t "image/*" --grant-read-uri-permission and it brings a standard ui menu to manually select home/lock.. That is not cool way, it should just change homescreen wallpaper without any confirmations.

Sounds like a pretty niche use, you can make any desired changed in the code or add your own custom intent support [here in AutoWallpaperService.kt](https://github.com/Hamza417/Peristyle/blob/master/app/src/main/java/app/simple/peri/services/AutoWallpaperService.kt). Auto Wallpaper service is designed to work in the background, or you can simply add your own service that takes all sorts of necessary Intents to do the different jobs. Peristyle so far only support [triggering auto wallpaper event using an Intent](https://github.com/Hamza417/Peristyle?tab=readme-ov-file#triggering-autowallpaperservice-from-other-apps) for [use cases like this](https://github.com/Hamza417/Peristyle/issues/22#issuecomment-2004438012).

**Comment by Hamza417** [comment]
I have added notifications that will show whenever the Auto Wallpaper service changes a wallpaper, along with a delete button separately for each of the screens. You can check the [screenshot here](https://github.com/Hamza417/Peristyle/blob/master/fastlane/metadata/android/en-US/images/phoneScreenshots/12.png).

**Comment by Hamza417** [comment]
> Hello. I used notifications solution a couple of days, it worked and it was ok, but than I decided working Termux-api is still needed and I flashed custom gsi rom and manage my wallpapers with termux now.

Is this related to the issue?
